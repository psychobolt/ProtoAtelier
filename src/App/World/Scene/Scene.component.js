// @flow
import React from 'react';
import {
  SceneModule,
  DefineModule,
  PerspectiveCamera,
  RenderingModule,
  OrbitControlsModule,
  ResizeModule,
} from 'whs/build/whs';
import { App } from 'react-whs';
import * as THREE from 'three';

type Props = {
  container: HTMLDivElement,
  width: number,
  height: number,
  style: {},
  children: any
};

export default class Scene extends React.Component<Props> {
  static defaultProps = {
    width: 680,
    height: 420,
  }

  constructor(props: Props) {
    super(props);
    const { width, height } = props;
    this.modules = [
      new SceneModule(),
      new DefineModule('camera', new PerspectiveCamera({
        aspect: width / height,
        position: new THREE.Vector3(0, 10, 50),
      })),
      new RenderingModule({
        bgColor: 0x162129,
        renderer: {
          antialias: true,
          shadowmap: {
            type: THREE.PCFSoftShadowMap,
          },
        },
        width,
        height,
      }, { shadow: true }),
      new OrbitControlsModule(),
    ];
  }

  modules: any[];

  render() {
    const { style, container, children } = this.props;
    return (
      <App
        modules={this.modules}
        parentStyle={style}
        passAppToView={({ native }) => {
          native.manager.set('container', container);
          native.applyModule(new ResizeModule());
        }}
      >
        {children}
      </App>
    );
  }
}
