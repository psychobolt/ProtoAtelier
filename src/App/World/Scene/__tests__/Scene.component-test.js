import React from 'react';
import { mount } from 'enzyme';

import { Scene } from '../Scene.component';

test('component <Scene /> should render without crashing', () => {
  mount(<Scene />);
});
