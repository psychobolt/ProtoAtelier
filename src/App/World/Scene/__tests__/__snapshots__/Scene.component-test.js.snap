// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Scene /> should render correctly 1`] = `
<Scene
  height={420}
  width={680}
>
  <t
    modules={
      Array [
        SceneModule {
          "scene": Object {
            "metadata": Object {
              "generator": "Object3D.toJSON",
              "type": "Object",
              "version": 4.5,
            },
            "object": Object {
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "type": "Scene",
              "uuid": 0,
            },
          },
        },
        DefineModule {
          "data": PerspectiveCamera$$1 {
            "_manager": ModuleManager {
              "currentModule": null,
              "handler": [Circular],
              "modules": Object {},
              "store": Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              },
            },
            "_native": Object {
              "metadata": Object {
                "generator": "Object3D.toJSON",
                "type": "Object",
                "version": 4.5,
              },
              "object": Object {
                "aspect": 1.619047619047619,
                "far": 1000,
                "filmGauge": 35,
                "filmOffset": 0,
                "focus": 10,
                "fov": 75,
                "matrix": Array [
                  1,
                  0,
                  0,
                  0,
                  0,
                  1,
                  0,
                  0,
                  0,
                  0,
                  1,
                  0,
                  0,
                  0,
                  0,
                  1,
                ],
                "near": 1,
                "type": "PerspectiveCamera",
                "uuid": 0,
                "zoom": 1,
              },
            },
            "_wait": Array [
              Promise {},
            ],
            "children": Array [],
            "emit": [Function],
            "modules": null,
            "off": [Function],
            "on": [Function],
            "params": Object {
              "aspect": 1.619047619047619,
              "build": true,
              "far": 1000,
              "fov": 75,
              "manager": true,
              "modules": null,
              "near": 1,
              "position": Vector3 {
                "x": 0,
                "y": 10,
                "z": 50,
              },
              "rotation": Object {
                "x": 0,
                "y": 0,
                "z": 0,
              },
            },
          },
          "name": "camera",
        },
        MockModule {},
        MockModule {},
      ]
    }
    parent="div"
    parentStyle={
      Object {
        "flex": 1,
      }
    }
    passAppToView={[Function]}
    start={true}
  >
    <div
      className="whs"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </t>
</Scene>
`;
